components/board/client.go

```diff
@@ -245,12 +249,52 @@ func (dic *digitalInterruptClient) Tick(ctx context.Context, high bool, nanoseco
 	panic(errUnimplemented)
 }
 
-func (dic *digitalInterruptClient) AddCallback(c chan Tick) {
+func (dic *digitalInterruptClient) AddCallback(ch chan Tick) {
 	panic(errUnimplemented)
 }
 
-func (dic *digitalInterruptClient) RemoveCallback(c chan Tick) {
-	panic(errUnimplemented)
+func (dic *digitalInterruptClient) RemoveCallback(ch chan Tick) {
+}
+
+func (c *client) StreamTicks(ctx context.Context, interrupts []string, ch chan Tick, extra map[string]interface{}) error {
+	ext, err := protoutils.StructToStructPb(extra)
+	if err != nil {
+		return err
+	}
+	stream := &interruptStream{
+		extra:  ext,
+		client: c,
+	}
+
+	c.mu.Lock()
+	c.interruptStreams = append(c.interruptStreams, stream)
+	c.mu.Unlock()
+
+	err = stream.startStream(ctx, interrupts, ch)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *client) removeStream(s *interruptStream) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for i := range s.interruptStreams {
```

Comment:
would this be cleaner with aa an `i, val` loop or do you care about the order? Totallay fine to optimize later.

Either way this particular code should be tested. it's smaller enough to create a list fo interrupt streams and call this function in a test.

Reply:
not sure how to call this function from the tests package without adding it to the board interface?

Reply:
make a client with the interrupt streams you need then call the function the remove what you need, then test that the stream you want removed has been removed
```go
testStreams := &interruptStreams{ what you need}
expectedStreams := &interruptStreams{after a removal}
 c := &client{
         streams : testStreams
 }
 c.removeStream(testStreams[oneOftem]
 testThat(t, c.streams, testShouldResemble(expectedStreams)
```

Reply:
oh I see, the client tests are part of board_test package. If you want add an interrupt_streams_test file in the board package, not board test to do this.

Reply:
A relevent reddit discussion https://www.reddit.com/r/golang/comments/twx02x/should_tests_and_mocks_be_in_a_test_package_or/

Reply:
The APIs and useability are encompassed  the `board_test` package, but since this is completely internal and we want a test for the logic, okay to write the test in the same package as `board`

Reply:
sounds good will do!